source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,233 @@
+filtering "green" documents out. The documents continue flowing through the pipeline and enter the ``$project`` stage,
+which transforms them from squares to circles. Lastly, the documents flow into the ``$group`` stage, configured to
+produce the ratio of red to blue circles.
+
+.. figure:: /figures/agg_factory.gif
+   :border:
+   :alt: Animation showing how the aggregation pipeline works, conceptually.
+
+
+Using the **Aggregation builders** API, the same operation may look something like:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/AggFactory.java
+   :language: java
+   :copyable: false
+   :dedent:
+   :start-after: begin aggregation
+   :end-before: end aggregation
+
+Which will produce output similar to the following:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id": 0, "red": 3, "blue": 2}
+
+Useful Links
+~~~~~~~~~~~~
+
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+- :ref:`Aggregation Builders <builders>`
+
+Runnable Examples
+-----------------
+
+Base Setup
+~~~~~~~~~~
+
+Create a new Java file called ``AggTour.java`` and include the following import statements:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/AggTour.java
+   :language: java
+   :dedent:
+   :start-after: begin imports
+   :end-before: end imports
+
+
+Connect to a MongoDB Deployment
++++++++++++++++++++++++++++++++
+
+.. code-block:: java
+
+   public class AggTour {
+
+       public static void main(String[] args) {
+           // Replace the uri string with your MongoDB deployment's connection string
+           String uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+
+           MongoClient mongoClient = MongoClients.create(uri);
+           MongoDatabase database = mongoClient.getDatabase("aggregation");
+           MongoCollection<Document> collection = database.getCollection("restaurants");
+
+           // aggregation here
+       }
+   }
+
+.. tip::
+
+   For information on connecting to MongoDB, see the :ref:`Connection Guide <mongoclient>`
+
+Insert the Data
++++++++++++++++
+
+.. literalinclude:: /includes/fundamentals/code-snippets/AggTour.java
+   :language: java
+   :dedent:
+   :start-after: begin insert
+   :end-before: end insert
+
+Basic Aggregation Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To perform an aggregation, pass a list of aggregation stages to the `MongoCollection.aggregate() <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/MongoCollection.html#aggregate(com.mongodb.client.ClientSession,java.util.List,java.lang.Class)>`__
+method.
+
+The Java driver provides the `Aggregates <https://mongodb.github.io/mongo-java-driver/4.2/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
```

Comment:
I feel like we should make clear somewhere that it's not required to use the builders.  Not sure whether this is the place though.  What do you think?

Reply:
I was going to frame this in the builders page, and frame it as "here's this manually constructed aggregation, here's the builder version".